app/Controllers/Http/ProfilesController.ts

```diff
@@ -0,0 +1,102 @@
+import { AuthenticationException } from '@adonisjs/auth/build/standalone';
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Profile from "App/Models/Profile";
+import Joi from 'joi';
+export default class ProfilesController {
+    async createprofile({ request, response, auth }: HttpContextContract) {
+        let userprofile = await Profile.findBy('mobile', request.body().mobile)
+        if (userprofile) {
+            return response.json({ message: "mobile number already exist" })
+        }
+
+        await this.validation(request.body())
+        const profile = request.all() as Partial<Profile>
+        const token = request.header('Authorization').replace('Bearer ', '')
+        const user = await auth.use('api').authenticate(token)
+        const userId = user.id
+        profile.user_id = userId;
+        const newprofile = await Profile.create(profile)
+        return newprofile;
+    }
+
+    async updateprofile({ request, response, auth }: HttpContextContract) {
+        const token = request.header('Authorization').replace('Bearer ', '')
+        try {
+            const user = await auth.use('api').authenticate(token)
```

Comment:
Same as above, Remove authentication from the controller. Also why we are not validating request body during update